flink-ml-lib/src/test/java/org/apache/flink/ml/classification/knn/KnnTest.java

```diff
@@ -0,0 +1,172 @@
+package org.apache.flink.ml.classification.knn;
+
+import org.apache.flink.api.common.restartstrategy.RestartStrategies;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.common.typeinfo.Types;
+import org.apache.flink.api.java.typeutils.RowTypeInfo;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.ml.api.core.Pipeline;
+import org.apache.flink.ml.api.core.Stage;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.environment.ExecutionCheckpointingOptions;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.api.functions.sink.SinkFunction;
+import org.apache.flink.table.api.Table;
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+import org.apache.flink.types.Row;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/** knn algorithm test. */
+public class KnnTest {
+    private StreamExecutionEnvironment env;
+    private StreamTableEnvironment tEnv;
+    private Table trainData;
+
+    List<Row> trainArray =
+            new ArrayList<>(
+                    Arrays.asList(
+                            Row.of("f", "2.0 3.0"),
+                            Row.of("f", "2.1 3.1"),
+                            Row.of("m", "200.1 300.1"),
+                            Row.of("m", "200.2 300.2"),
+                            Row.of("m", "200.3 300.3"),
+                            Row.of("m", "200.4 300.4"),
+                            Row.of("m", "200.4 300.4"),
+                            Row.of("m", "200.6 300.6"),
+                            Row.of("f", "2.1 3.1"),
+                            Row.of("f", "2.1 3.1"),
+                            Row.of("f", "2.1 3.1"),
+                            Row.of("f", "2.1 3.1"),
+                            Row.of("f", "2.3 3.2"),
+                            Row.of("f", "2.3 3.2"),
+                            Row.of("c", "2.8 3.2"),
+                            Row.of("d", "300. 3.2"),
+                            Row.of("f", "2.2 3.2"),
+                            Row.of("e", "2.4 3.2"),
+                            Row.of("e", "2.5 3.2"),
+                            Row.of("e", "2.5 3.2"),
+                            Row.of("f", "2.1 3.1")));
+
+    List<Row> testArray =
+            new ArrayList<>(Arrays.asList(Row.of("e", "4.0 4.1"), Row.of("m", "300 42")));
+
+    private Table testData;
+
```

Comment:
Can we add more tests for functions like below? Examples can be found in `KmeansTest`.

- test get/setModelData
- test params

Reply:
OK, I  will do it later.

Reply:
done